src/util/psbt/macros.rs

```diff
@@ -136,7 +136,7 @@ macro_rules! impl_psbt_get_pair {
             });
         }
     };
-    ($rv:ident.push($slf:ident.$keyed_name:ident as <$keyed_typeval:expr, $keyed_key_type:ty>|<$keyed_value_type:ty>)) => {
+    ($rv:ident.push($slf:ident.$keyed_name:ident as BTreeMap<$keyed_typeval:ident>)) => {
```

Comment:
We can also remove the BTreeMap here, and change this to `rv.push` and the above can be `rv.push_map`. The above types `Option` and `BTreeMap` are merely matching the syntax and not used in the macro.

If you do the above suggestions, the code might look simpler:

```rust
            rv.push(self.witness_script as PSBT_OUT_WITNESS_SCRIPT)
            rv.push_map(self.bip32_derivation as PSBT_OUT_BIP32_DERIVATION)
```